config/prometheus-adapter-helm-values.yaml

```diff
@@ -11,7 +11,8 @@ rules:
       overrides:
         kubernetes_namespace: {resource: "namespace"}
         kubernetes_pod_name: {resource: "pod"}
+        name: {group: "apps", resource: "deployment"}
     name:
       matches: "num_ready_work_items"
-      as: "num_ready_work_items_avg"
-    metricsQuery: 'avg_over_time(<<.Series>>{<<.LabelMatchers>>}[2m])'
+      as: "num_ready_work_items"
+    metricsQuery: '<<.Series>>{<<.LabelMatchers>>}'
```

Comment:
The HPA has a built in stabilization window to prevent thrashing https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#stabilization-window so I figured we could use that instead of relying on the time averaged metric.

We're still using an average elsewhere however (`averageValue`), but as part of the HPA definition, so that the HPA averages the Prometheus metric over the number of pods currently running.

Reply:
I'm not quite sure what 'averages over the number of pods' means. Each pod (for a given service) reports exactly the same value of `num_ready_work_items`, so averaging would be pointless. Unless I'm misunderstanding something. But if it works I'm ok with it.

Reply:
Since we're using `averageValue` in the HPA definition, and our Prometheus query will return a single metric value per deployment, the HPA will calculate its metric using `num_ready_work_items / number of pods`, similar to what this example describes: 

https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/#autoscaling-on-metrics-not-related-to-kubernetes-objects

I should've stated that in a different way, but hopefully that makes more sense.